hibernate-core/src/test/java/org/hibernate/orm/test/jpa/lock/RepeatableReadTest.java

```diff
@@ -239,41 +243,48 @@ public void testStaleNonVersionedInstanceFoundOnLock() {
 		Long partId = p.getId();
 
 		// Now, open a new Session and re-load the part...
-		inTransaction(
-				s1 -> {
-					Part part = s1.get( Part.class, partId );
+		// TiDB will throw a TransactionException by writing conflict at commit
+		try {
+			inTransaction(
+					s1 -> {
+						Part part = s1.get( Part.class, partId );
 
-					// now that the item is associated with the persistence-context of that session,
-					// open a new session and modify it "behind the back" of the first session
-					inTransaction(
-							s2 -> {
-								Part part2 = s2.get( Part.class, partId );
-								part2.setName( "Lock Mode Types" );
-							}
-					);
-
-					// at this point, s1 now contains stale data, so acquire a READ lock
-					// and make sure we get the already associated state (i.e., the old
-					// name and the old version)
-					s1.lock( part, LockMode.READ );
-					Part part2 = s1.get( Part.class, partId );
-					assertTrue( part == part2 );
-					assertEquals( check, part2.getName(), "encountered non-repeatable read" );
+						// now that the item is associated with the persistence-context of that session,
+						// open a new session and modify it "behind the back" of the first session
+						inTransaction(
+								s2 -> {
+									Part part2 = s2.get( Part.class, partId );
+									part2.setName( "Lock Mode Types" );
+								}
+						);
 
-					// then acquire an UPGRADE lock; this should fail
-					try {
-						s1.lock( part, LockMode.PESSIMISTIC_WRITE );
-					}
-					catch (Throwable t) {
-						// SQLServer, for example, immediately throws an exception here...
-						s1.getTransaction().rollback();
-						s1.beginTransaction();
+						// at this point, s1 now contains stale data, so acquire a READ lock
+						// and make sure we get the already associated state (i.e., the old
+						// name and the old version)
+						s1.lock( part, LockMode.READ );
+						Part part2 = s1.get( Part.class, partId );
+						assertTrue( part == part2 );
+						assertEquals( check, part2.getName(), "encountered non-repeatable read" );
+
+						// then acquire an UPGRADE lock; this should fail
+						try {
+							s1.lock( part, LockMode.PESSIMISTIC_WRITE );
+						}
+						catch (Throwable t) {
+							// SQLServer, for example, immediately throws an exception here...
+							s1.getTransaction().rollback();
+							s1.beginTransaction();
+						}
+						part2 = s1.get( Part.class, partId );
+						assertTrue( part == part2 );
+						assertEquals( check, part2.getName(), "encountered non-repeatable read" );
 					}
-					part2 = s1.get( Part.class, partId );
-					assertTrue( part == part2 );
-					assertEquals( check, part2.getName(), "encountered non-repeatable read" );
-				}
-		);
+			);
+		} catch (TransactionException e) {
+			if (!(getDialect() instanceof TiDBDialect)) {
+				throw e;
+			}
+		}
```

Comment:
Please revert these changes since you added a `@SkipDialect` anyway

Reply:
Reverted